client/my-sites/plugins/plugin-meta/index.jsx

```diff
@@ -170,7 +170,7 @@ export default React.createClass( {
 				text={ i18n.translate( 'A new version is available.' ) }
 				status="is-warning"
 				button={ i18n.translate( 'Update to %(newPluginVersion)s', { args: { newPluginVersion: newVersion } } ) }
-				onClick={ this.handlePluginUpdates }
+				onDismissClick={ this.handlePluginUpdates }
 				showDismiss={ false } />;
```

Comment:
Since `showDismiss` is false here, `onDismissClick` will never be fired. I do believe the component wants to trigger `handlePluginUpdates` when the child link is clicked the `Update to 3.8.1` link, so this should be worked around before merging.